crop_classification/inference.py

```diff
@@ -0,0 +1,111 @@
+from cropharvest.engineer import Engineer
+from datetime import datetime
+from pathlib import Path
+import numpy as np
+import xarray as xr
+import pandas as pd
+import re
+import torch
+
+from typing import Dict, List, Optional, Tuple, Union
+
+
+class Inference:
+    """
+    Class for running inference using either a pytorch checkpoint model or
+    a pytorch jit model on a single tif file.
+    """
+
+    def __init__(self, model, device: Optional[torch.device] = None):
+        self.model = model
+        self.device = device
+
+        # TODO: normalizing_dict is property of the model in crop-mask should this be the same here?
+        self.normalizing_dict: Dict[str, np.ndarray] = None
+
+        # TODO: batch size is a property of the model in crop-mask should this be the same here?
+        self.batch_size: int = 64
+
+    @staticmethod
+    def start_date_from_str(path: Union[Path, str]) -> datetime:
+        dates = re.findall(r"\d{4}-\d{2}-\d{2}", str(path))
+        if len(dates) != 2:
+            raise ValueError(f"{path} should have start and end date")
+        start_date_str, _ = dates
+        start_date = datetime.strptime(start_date_str, "%Y-%m-%d")
+        return start_date
+
+    @staticmethod
+    def _tif_to_np(
+        local_path: Path,
+        start_date: datetime,
+        normalizing_dict: Optional[Dict[str, np.ndarray]] = None,
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+
+        # TODO: This is close to Engineer().process_test_file()
+        # but does not rely on the Engineer class being initialized with labels
+        # might make sense to add a majority of this class to cropharvest directly
```

Comment:
@gabrieltseng Thoughts about this comment?

Reply:
hmm. do you mean coupling the inference class to cropharvest?

I would prefer to keep it here, since its a specific application of the dataset. What might make sense is to move this function into the engineer, as a class method

```python
# (in Engineer)
     @classmethod
     def tif_to_np(cls, local_path, start_date):
         da, slope = cls.load_tif(local_path, start_date)
        ...
```
Then, we can just call that function here, and do the same in `process_test_file`

Reply:
> do you mean coupling the inference class to cropharvest?
Yes

> I would prefer to keep it here, since its a specific application of the dataset

I think the this `Inference` class actually should work with any application of the dataset. I made an almost identical one for crop-mask. It essentially just simplifies running a cropharvest exported tif through a model trained on cropharvest data.

Reply:
hmm, yes actually good point. I think it makes sense for this to be moved into the cropharvest repo then

Reply:
Moved in this PR: https://github.com/nasaharvest/cropharvest/pull/75